### PR TITLE
Compatible with jruby

### PR DESCRIPTION
**Pull request by wokenqingtian**
Base on #283  Better merges it first.

#### What does this PR do?

- Update development configuration to support JRuby
- Use docker-compose and Make to do local multiple Ruby versions development
- Add JRuby to Travis CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,37 +1,18 @@
...
 rvm:
-  - 2.1.10
-  - 2.2.1
-  - 2.2.2
-  - 2.2.10
-  - 2.3.8
-  - 2.4.9
   - 2.5.7
   - 2.6.5
   - 2.7.0
```
Comment (runlevel5) [inline .travis.yml:6]: What do you think about testing the latest minor version instead of explicit version?

File: .travis.yml [hunk 2]
```diff
@@ -1,37 +1,18 @@
...
-before_install:
-  - gem install bundler -v '~> 1.0'
+  - jruby-head
```
Comment (runlevel5) [inline .travis.yml:7]: Is there any reason why we want to test the jruby-head here? Should we test against jruby 9.2?

File: .travis.yml (hunk 1 above)
Comment (wokenqingtian) [reply .travis.yml:6]: Good idea. I will do that

Comment (wokenqingtian) [reply .travis.yml]: I got a wired result:

* [2.5 points to 2.5.5](https://travis-ci.org/github/jeremytregunna/ruby-trello/jobs/681528450)
* [2.6 points to 2.6.3](https://travis-ci.org/github/jeremytregunna/ruby-trello/jobs/681528453)

Maybe I should change back. Or did I do something wrong?

File: .travis.yml (hunk 2 above)
Comment (wokenqingtian) [reply .travis.yml]: [It seems that JRuby 9.2 is working on Travis.](https://travis-ci.org/github/jeremytregunna/ruby-trello/jobs/681528458)

File: .travis.yml (hunk 1 above)
Comment (runlevel5) [reply .travis.yml]: @wokenqingtian let's report that to Travis CI team instead

Comment (wokenqingtian) [reply .travis.yml]: OK, sure

**Comment by jeremytregunna** [comment]
> LGTM +1

Me too!

File: .travis.yml (hunk 1 above)
Comment (wokenqingtian) [reply .travis.yml]: After report this issue, Travis team [fixed it](https://github.com/travis-ci/travis-build/pull/1902) immediately. I also checked our recent CI logs. All point to correct Ruby versions.

**Pull request merged**